openem-ingestor/core/config.go

```diff
@@ -0,0 +1,62 @@
+package core
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/spf13/viper"
+)
+
+type Scicat struct {
+	Host        string `string:"Host"`
+	AccessToken string `string:"AccessToken"`
+}
+
+type S3 struct {
+	Endpoint string `string:"Endpoint"`
+	Bucket   string `string:"Bucket"`
+	Checksum bool   `bool:"Checksum"`
+}
+
+type Globus struct {
+	Endpoint string `string:"Endpoint"`
+}
+
+type Transfer struct {
+	Method string `string:"Method"`
+	S3     S3     `mapstructure:"S3"`
+	Globus Globus `mapstructure:"Globus"`
+}
+
+type Misc struct {
+	ConcurrencyLimit int `int:"ConcurrencyLimit"`
+}
+
+type Config struct {
+	Scicat   Scicat   `mapstructure:"Scicat"`
+	Transfer Transfer `mapstructure:"Transfer"`
+	Misc     Misc     `mapstructure:"Misc"`
+}
+
+func GetConfig() (Config, error) {
+	var config Config
+	if err := viper.Unmarshal(&config); err != nil {
+		fmt.Println(err)
+		return config, err
+	}
+	return config, nil
+}
+
+func ReadConfig() error {
+	viper.SetConfigName("openem-ingestor-config") // name of config file (without extension)
+	viper.SetConfigType("yaml")
+
+	userConfigDir, _ := os.UserConfigDir()
+
+	viper.AddConfigPath("./")
```

Comment:
forward slash only works on non-Windows systems and if the PWD is elsewhere it will not work anywhere. The PWD could be elsewhere if the user runs the executable from a different directory like so: `some/path/to/executable/ingestor-app`. It's better to use `os.Executable()` with `path.Dir()` to get an absolute path to the executable's directory